File: 3005-count-elements-with-maximum-frequency/3005-count-elements-with-maximum-frequency.cpp
class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        unordered_map<int, int> map;
        for(int i = 0; i<nums.size(); i++){
            map[nums[i]]++;
        }
        
        int maxFreq = 0;
        for(auto ele:map){
            maxFreq = max(maxFreq, ele.second);
        }
        // for(int i = 0; i<map.size(); i++){
        //     maxFreq = max(maxFreq, map[nums[i]].second());
        // }
        
        int res = 0;
        for(auto ele:map){
            if(ele.second == maxFreq)    res += maxFreq;
        }
        return res;
    }
};

File: 3005-count-elements-with-maximum-frequency/3005-count-elements-with-maximum-frequency.py
class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        dic = {}
        for i in nums:
            if(i in dic):
                dic[i]+=1
            else:
                dic[i] = 1
        
        maxFreq = 0
        
        for i in dic:
            if(dic[i] > maxFreq):
                maxFreq = dic[i]
        
        res = 0
        for i in dic:
            if(dic[i] == maxFreq):
                res += maxFreq
        
        return res
        